app/views/collections/show.html.erb

```diff
@@ -0,0 +1,186 @@
+    color: #bb6c6c;
+  }
+</style>
+<h1><%= @collection.title.strip %></h1>
+
+<p>Code: <%= @collection.code %></p>
+<p>Description: <%= @collection.description %></p>
+
+
+<h2>Submitters</h2>
+<p>Users that are allowed to add new works to this collection</p>
+<ul id="submitter-list">
+<% @collection.submitters.each do |user| %>
+  <!--
+    Notice that we render the list of submitters and admins as hidden spans that are then
+    picked up by the JavaScript on the page and turned into visible HTMLs with all the bells
+    and whistles. This is so that we have a single way of rendering the data (both the user
+    permissions already on the database *and* the user permissions added on the fly by an
+    administrator).
+  -->
+  <span class="hidden submitter-data"
+    data-uid="<%= user.uid %>"
+    data-collection-id="<%= @collection.id %>"
+    data-can-delete="<%= @can_edit && user.uid != current_user.uid %>"
+    data-you="<%= user.uid == current_user.uid %>"></span>
+<% end %>
+</ul>
+
+<% if @can_edit == true %>
+  <div>
+      <input id="submitter-uid-to-add" placeholder="netid"></input>
+      <%= link_to "Add Submitter", "#", id: "btn-add-submitter", class: "btn btn-secondary" %>
+      <span id="add-submitter-message"></span>
+  </div>
+<% end %>
+
+
+<h2>Administrators</h2>
+<p>Users that are allowed to admininister this collection</p>
+<ul id="admin-list">
+<% @collection.administrators.each do |user| %>
+  <span class="hidden admin-data"
+    data-uid="<%= user.uid %>"
+    data-collection-id="<%= @collection.id %>"
+    data-can-delete="<%= @can_edit && user.uid != current_user.uid %>"
+    data-you="<%= user.uid == current_user.uid %>"></span>
+<% end %>
+</ul>
+
+<% if @can_edit == true %>
+  <div>
+      <input id="admin-uid-to-add" placeholder="netid"></input>
+      <%= link_to "Add Administrator", "#", id: "btn-add-admin", class: "btn btn-secondary" %>
+      <span id="add-admin-message"></span>
+  </div>
+<% end %>
+
+<% if @can_edit == true %>
+  <div>
+    <%= link_to("Edit", edit_collection_path(@collection), class: "btn btn-primary") %>
+  </div>
+<% end %>
+
+<%= link_to("View all collections", collections_path) %>
+
+<script>
+$(function() {
+
+  // Issues the HTTP DELETE to remove a user's access from a collection
+  var deleteUserFromCollection = function(el, url, uid) {
+    $.ajax({
+      type: "DELETE",
+      url: url.replace("uid-placeholder", uid),
+      data: { authenticity_token: "<%= form_authenticity_token %>" },
+      success: function() {
+        // Remove the <li> for the user
+        $(".li-user-" + uid).remove();
+      },
+      error: function(x) {
+        alert(x.responseJSON.message);
+      }
+    });
+  }
+
+  // Adds the information for a given user to the lists of administrators/submitters
+  // for the collection.
+  // `elList` is the reference to the <ul> HTML element that hosts the list.
+  var addUserHtml = function(elList, uid, collectionId, role, canDelete, isYou) {
```

Comment:
It might be a bit unorthodox to add the list of submitters and administrators on the page via JavaScript, but with this approach we have a *single logic* to display this information regardless of whether we are displaying the current data on the database vs the new records added on the fly by the user (this is in contrast to displaying the information sometimes with Ruby code and other times with JavaScript code.)